src/test/java/seedu/address/logic/parser/BuyCommandParserTest.java

```diff
@@ -0,0 +1,87 @@
+
+    private static final String VALID_GOOD_NAME_DESC = " " + PREFIX_GOOD_NAME + VALID_GOOD_NAME;
+    private static final String VALID_GOOD_NAME_EXTRA_DESC = " " + PREFIX_GOOD_NAME + VALID_GOOD_NAME_EXTRA;
+    private static final String INVALID_GOOD_NAME_DESC = " " + PREFIX_GOOD_NAME + INVALID_GOOD_NAME;
+    private static final String VALID_GOOD_QUANTITY_DESC = " " + PREFIX_QUANTITY + VALID_GOOD_QUANTITY;
+    private static final String VALID_GOOD_QUANTITY_EXTRA_DESC = " " + PREFIX_QUANTITY + VALID_GOOD_QUANTITY_EXTRA;
+    private static final String INVALID_GOOD_QUANTITY_NEGATIVE_DESC =
+            " " + PREFIX_QUANTITY + INVALID_GOOD_QUANTITY_NEGATIVE;
+    private static final String INVALID_GOOD_QUANTITY_OVERFLOW_INT_DESC =
+            " " + PREFIX_QUANTITY + INVALID_GOOD_QUANTITY_OVERFLOW_INT;
+
+    private BuyCommandParser parser = new BuyCommandParser();
+    private Good validGood = new Good(new GoodName(VALID_GOOD_NAME), new GoodQuantity(VALID_GOOD_QUANTITY));
+
+    @Test
+    void parse_allFieldsPresent_success() {
+        BuyCommand expectedCommand = new BuyCommand(validGood);
+        // Happy case
+        assertParseSuccess(parser, VALID_GOOD_NAME_DESC + VALID_GOOD_QUANTITY_DESC,
+                new BuyCommand(validGood));
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + VALID_GOOD_NAME_DESC + VALID_GOOD_QUANTITY_DESC,
+                new BuyCommand(validGood));
+
+        // multiple goodNames, only last one accepted
+        assertParseSuccess(parser, VALID_GOOD_NAME_EXTRA_DESC + VALID_GOOD_NAME_DESC
+                + VALID_GOOD_QUANTITY_DESC, new BuyCommand(validGood));
+
+        // multiple quantities, only last one accepted
+        assertParseSuccess(parser, VALID_GOOD_NAME_DESC + VALID_GOOD_QUANTITY_EXTRA_DESC
+                + VALID_GOOD_QUANTITY_DESC, new BuyCommand(validGood));
+    }
+
+    @Test
+    void parse_compulsoryFieldMissing_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, BuyCommand.MESSAGE_USAGE);
+
+        // missing good name
+        assertParseFailure(parser, VALID_GOOD_QUANTITY_DESC,
+                expectedMessage);
+
+        // missing good quantity
+        assertParseFailure(parser, VALID_GOOD_NAME_DESC,
+                expectedMessage);
+    }
+
+    @Test
+    void parse_invalidValue_failure() {
+        // invalid good name
+        assertParseFailure(parser, INVALID_GOOD_NAME_DESC + VALID_GOOD_QUANTITY_DESC,
+                GoodName.MESSAGE_CONSTRAINTS);
+
+        // invalid good quantities
+        assertParseFailure(parser, VALID_GOOD_NAME_DESC + INVALID_GOOD_QUANTITY_NEGATIVE_DESC,
+                GoodQuantity.MESSAGE_CONSTRAINTS);
+        /*
+        TODO: discuss with group mates how to solve int overflow problem.
+        assertParseFailure(parser, VALID_GOOD_NAME_DESC + INVALID_GOOD_QUANTITY_OVERFLOW_INT_DESC,
+                GoodQuantity.MESSAGE_CONSTRAINTS);
+         */
```

Comment:
For me, I put limits on the digit in the regex

Reply:
Good idea. Limited regex string to length 6, i.e. value of 999,999 max